code-challenges.test.js

```diff
@@ -14,38 +14,149 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
-const people = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
+describe("sentenceGenerator", () => {
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized.", () => {
+    const people = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+      { name: "arthur dent", occupation: "a radio employee" }
+    ]
+    expect(sentenceGenerator(people)).toEqual(["Ford Prefect is a hitchhiker", "Zaphod Beeblebrox is president of the galaxy", "Arthur Dent is a radio employee"])
+  })
+})
+
+// JEST "GOOD" FAILURE
+
+//  ● sentenceGenerator › takes in an array of objects and returns an array with a sentence about each person with their name capitalized.
+
+//   ReferenceError: sentenceGenerator is not defined
+
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
 
-// b) Create the function that makes the test pass.
+
+// b) Create the function that makes the test pass. 
+
+    // *** STUDENT NOTE **** 
+
+    //  I used quite a bit more pseudo code than I typically would. This problem was quite challenging for me to wrap my head around programatically. So, please excuse the 'wordy' pseudo code.
+
+    // Create FUNCTION "sentenceGenerator" that takes in ARRAY
+    // Need to RETURN the passed in array and .MAP over each Array(VALUE) to .split the names
+    // So the .map HOF needs to Split on the VALUE.NAME element on each Array element so I can target the 1st letters in the names next <-- <-- (set as VAR "splitArray")
+    // Now take the newly created array of split values (as VAR uppercaseArray) and .toUpperCase the names at the indexes **[0][0]** concatnate with + and .substring to bring letters/words back together. 
+    // Should be able to now RETURN via STRING INTERPOLATION to bring the newly capitalized NAMES into a sentence with corresponding occupation VALUES.
+
+const sentenceGenerator = (array) => {
+  return array.map((value) => {
+    let splitArray = value.name.split(' ')
+    let uppercaseArray = splitArray[0][0].toUpperCase() + splitArray[0].substring(1) + " " + splitArray[1][0].toUpperCase() + splitArray[1].substring(1)
+    return `${uppercaseArray} is ${value.occupation}`
+  })
+}
```

Comment:
Nice!  This problem is definitely a tricky one!  You could also do a nested map and iterate on the split name for a more dynamic approach to this problem.